nodebuilder/p2p/bitswap.go

```diff
@@ -29,16 +31,38 @@ const (
 // dataExchange provides a constructor for IPFS block's DataExchange over BitSwap.
 func dataExchange(params bitSwapParams) exchange.Interface {
 	prefix := protocol.ID(fmt.Sprintf("/celestia/%s", params.Net))
```

Comment:
it'd be nice to have this as a func on bitSwapParams so you could spit out `params.ProtococID()` and we could write a little unit test for it, maybe even add `params.NeworkPrefix()` and we can test both and then just pass that as 3rd arg to `network.NewFromIpfsHost`

Reply:
I see your point. Not sure why that should be a method on bitSwapParams, though we could also extract that in a func and similarly unit test it. I don't think that the params struct needs any methods

Reply:
Extract in a func

Reply:
doing too many years of OOP is why i have the impulse to put it ON something, but func works exactly the same 👍